src/openshift/cluster.ts

```diff
@@ -67,12 +65,16 @@ export class Cluster extends OpenShiftItem {
         }
     }
 
-    private static async loginDialog(): Promise<string> {
-        const clusterURL = await vscode.window.showInputBox({
+    static async getUrl() {
+        return await vscode.window.showInputBox({
             ignoreFocusOut: true,
             prompt: "Provide URL of the cluster to connect",
             validateInput: (value: string) => Cluster.validateUrl('Invalid URL provided', value)
         });
+    }
+
+    private static async loginDialog(): Promise<string> {
+        const clusterURL = await Cluster.getUrl();
```

Comment:
Should work the same way we have done for `Create Component` command.  First it should ask for login method, then call specific commands to login with credentials or with token.
`const clusterURL = await Cluster.getUrl();` call should be part for those commands.

Reply:
I think it's  better to keep `const clusterURL = await Cluster.getUrl();` here only because if I call credentialsLogin/tokenLogin command from the `log in icon view/command palette` then I don't have any condition to check whether I am running from command palette or not. https://github.com/redhat-developer/vscode-openshift-tools/pull/681/files#diff-acaf210d416301d185dd5622c49f9093R89 will run twice if I try to login from the icon view and I want to show this first when the user tries to login from the login icon view.

Reply:
let's say we have two different commands. One does login with credentials and another with token.
Each command would do:
* check for user being logged in
* ask for cluster URL
* ask credentials for specific login type: username/password or token.
Each command would have a parameter `do not check login status'.

When you run from command palette we pass values for parameter 'do not check login status' and in this case each command supposed to run check.

Now we need command that asks what login do you want to use username/password or token. It would do:
* check for login status and ask confirmation for login to different cluster
* show quick picks for username/password and token logins
* call specific login function directly with 'do not check login status' parameter value set to `true'

That would be much cleaner workflow.